ambry-store/src/test/java/com.github.ambry.store/CompactionManagerTest.java

```diff
@@ -0,0 +1,291 @@
+    verifyCompactionDetails(null);
+
+    // 1 valid LogSegment
+    blobStore.validLogSegments = generateRandomString(1);
+    verifyCompactionDetails(new CompactionDetails(blobStore.getTime().milliseconds() - messageRetentionTimeInMs,
+        blobStore.validLogSegments));
+
+    // random no of valid logSegments
+    for (int i = 0; i < 3; i++) {
+      int logSegmentCount = TestUtils.RANDOM.nextInt(10) + 1;
+      blobStore.validLogSegments = generateRandomString(logSegmentCount);
+      verifyCompactionDetails(new CompactionDetails(blobStore.getTime().milliseconds() - messageRetentionTimeInMs,
+          blobStore.validLogSegments));
+    }
+  }
+
+  /**
+   * Tests {@link CompactionManager#getCompactionDetails(BlobStore)} for different values for {@link BlobStore}'s
+   * used capacity
+   * @throws StoreException
+   */
+  @Test
+  public void testDifferentUsedCapacities() throws StoreException {
+    // mingLogSize = 50%
+    // different valid used capacity
+    blobStore.validLogSegments = generateRandomString(2);
+    // if used capacity fi less than 60%, compaction details will be null
+    Long[] usedCapacities =
+        new Long[]{CAPACITY_IN_BYTES * 2 / 10, (CAPACITY_IN_BYTES * 4 / 10), CAPACITY_IN_BYTES * 5 / 10};
+    for (Long usedCapacity : usedCapacities) {
+      blobStore.usedCapacity = usedCapacity;
+      verifyCompactionDetails(null);
+    }
+
+    // if used capacity is greater than 60%, compaction details will be non null
+    usedCapacities = new Long[]{
+        CAPACITY_IN_BYTES * 51 / 100, (CAPACITY_IN_BYTES * 6 / 10),
+        CAPACITY_IN_BYTES * 7 / 10, CAPACITY_IN_BYTES * 9 / 10};
+    for (Long usedCapacity : usedCapacities) {
+      blobStore.usedCapacity = usedCapacity;
+      verifyCompactionDetails(new CompactionDetails(blobStore.getTime().milliseconds() - messageRetentionTimeInMs,
+          blobStore.validLogSegments));
+    }
+  }
+
+  /**
+   * Tests {@link CompactionManager#getCompactionDetails(BlobStore)} for different values for
+   * {@link StoreConfig#storeMinLogSizeToTriggerCompactionInPercent}
+   */
+  @Test
+  public void testDifferentThresholdsForMinLogSizeToCompact() throws StoreException {
+    int[] minLogSizeToTriggerCompactionInPercentages = new int[]{60, 61, 65, 70, 80, 95};
+    // when used capacity is 60%, compactionDetails is expected to be null for all of the above values
+    for (int minLogSize : minLogSizeToTriggerCompactionInPercentages) {
+      properties.setProperty("store.min.log.size.to.trigger.compaction.in.percent", String.valueOf(minLogSize));
+      StoreConfig config = new StoreConfig(new VerifiableProperties(properties));
+      MetricRegistry metricRegistry = new MetricRegistry();
+      StorageManagerMetrics metrics = new StorageManagerMetrics(metricRegistry);
+      blobStore = new MockBlobStore("", config, null, null, metrics, null, 0, null, null, null, time);
+      blobStore.capacityInBytes = CAPACITY_IN_BYTES;
+      blobStore.start();
```

Comment:
the start calls might not be required since you are mocking the `BlobStore`, right?